storage: add comment to ReplicaPlaceholder

tbg opened the pull request:
It deserves one, took me a minute and some git blame'ing to remember
what this was all about.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/38331)
<!-- Reviewable:end -->

tbg commented on pkg/storage/replica_placeholder.go:
```diff
@@ -19,8 +19,52 @@ import (
+// are kept synchronously in two places in (*Store).mu, namely the
+// replicaPlaceholders and replicaByKey maps.
+//
+// Placeholders are inserted in `(*Store).processRaftSnapshotRequest)` when a
```
Jeez, this sentence is quite the sandwich. I'm going to leave as is and address it with the review cycle

danhhz reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38331#-:-Lhqe7IG4s2RBwfCFOJP:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @danhhz and @tbg)

___
*[pkg/storage/replica_placeholder.go, line 22 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38331#-LhqeCCJ1VRe8aW2uWl7:-LhqeCCJ1VRe8aW2uWl8:blmd03h) ([raw file](https://github.com/cockroachdb/cockroach/blob/3707597a44d2fd2aa3a8c7623b2915bfdc758826/pkg/storage/replica_placeholder.go#L22)):*
> ```Go
> )
> 
> // ReplicaPlaceholder represents a "lock" of a part of the keyspace on a given
> ```

as I mentioned over slack, I don't love the name ReplicaPlaceholder. something like StoreKeyspaceLock might be more illustrative. this definitely increases the scope of this PR and I don't want to punish a good deed, so absolutely feel free to ignore this suggestion
___
*[pkg/storage/replica_placeholder.go, line 27 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38331#-Lhq_62J00TXGrWDXjl_-r1-27:-LhqdPFU9A_wSZLAfbny:b-cotoao) ([raw file](https://github.com/cockroachdb/cockroach/blob/3707597a44d2fd2aa3a8c7623b2915bfdc758826/pkg/storage/replica_placeholder.go#L27)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Jeez, this sentence is quite the sandwich. I'm going to leave as is and address it with the review cycle
</blockquote></details>

heh, yeah. i can't parse this paragraph at all. when you rework it, i'd move it to the bottom of this comment. it's much easier to understand the "how" after you've gone over the "why"

you could also just kill this paragraph completely. i'm not sure it adds much and it's much more prone to rot than the rest of this
___
*[pkg/storage/replica_proposal.go, line 1014 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38331#-LhqdNZl3PplQHradHSH:-LhqdNZm4mwpFG7CBUxt:b-7h5vzi) ([raw file](https://github.com/cockroachdb/cockroach/blob/3707597a44d2fd2aa3a8c7623b2915bfdc758826/pkg/storage/replica_proposal.go#L1014)):*
> ```Go
> 		Request: &ba,
> 	}
> 	clusterVersionActive := true
> ```

unrelated?


<!-- Sent from Reviewable.io -->

tbg committed:
storage: add comment to ReplicaPlaceholder

It deserves one, took me a minute and some git blame'ing to remember
what this was all about.

Release note: None

tbg reviewed (commented):
TFTR!

bors r=danhhz

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38331#-:-LhqnhgpCAElLLTp80q-:b-vwgs12)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @danhhz)

___
*[pkg/storage/replica_placeholder.go, line 22 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38331#-LhqeCCJ1VRe8aW2uWl7:-LhqlgHAE1qm2gmHSSwF:b-kq3otd) ([raw file](https://github.com/cockroachdb/cockroach/blob/3707597a44d2fd2aa3a8c7623b2915bfdc758826/pkg/storage/replica_placeholder.go#L22)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

as I mentioned over slack, I don't love the name ReplicaPlaceholder. something like StoreKeyspaceLock might be more illustrative. this definitely increases the scope of this PR and I don't want to punish a good deed, so absolutely feel free to ignore this suggestion
</blockquote></details>

It's not ignored, I just want to leave it out of this PR - removing the term "placeholder" requires some auditing and random changes in other comments, so I don't want to lump it in here. Let's mull over the name some more before we commit.
___
*[pkg/storage/replica_placeholder.go, line 27 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38331#-Lhq_62J00TXGrWDXjl_-r1-27:-LhqlqSVC_ns3rbILjtU:bnsugoj) ([raw file](https://github.com/cockroachdb/cockroach/blob/3707597a44d2fd2aa3a8c7623b2915bfdc758826/pkg/storage/replica_placeholder.go#L27)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

heh, yeah. i can't parse this paragraph at all. when you rework it, i'd move it to the bottom of this comment. it's much easier to understand the "how" after you've gone over the "why"

you could also just kill this paragraph completely. i'm not sure it adds much and it's much more prone to rot than the rest of this
</blockquote></details>

Ok, removed. I touched up some other comments to make up for it.
___
*[pkg/storage/replica_proposal.go, line 1014 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38331#-LhqdNZl3PplQHradHSH:-LhqnV9YD-6aPQjJSNog:b-3fhcvj) ([raw file](https://github.com/cockroachdb/cockroach/blob/3707597a44d2fd2aa3a8c7623b2915bfdc758826/pkg/storage/replica_proposal.go#L1014)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

unrelated?
</blockquote></details>

Super unrelated, not sure how that slipped in.


<!-- Sent from Reviewable.io -->

tbg commented:
Known sql flake https://github.com/cockroachdb/cockroach/issues/38062

bors r=danhhz
